synapse/metrics/__init__.py

```diff
@@ -68,9 +68,18 @@ def register_cache(self, *args, **kwargs):
 
 
 def register_memory_metrics(hs):
-    metric = MemoryUsageMetric(hs)
+    try:
+        import psutil
+        process = psutil.Process()
+        process.memory_info().rss
+    except (ImportError, AttributeError):
+        logger.warn(
+            "psutil is not installed or incorrect version."
+            " Disabling memory metrics."
+        )
+        return
```

Comment:
I prefer to keep `try...except` as narrow as possible, to ensure we don't accidentally consume legitimate errors.